scripts/apps/authoring-react/authoring-integration-wrapper.tsx

```diff
@@ -185,16 +184,20 @@ const getExportModal = (
 const getHighlightsAction = (getItem: () => IArticle): IAuthoringAction => {
     return {
         label: gettext('Highlights'),
-        onTrigger: () => (
-            showModal(({closeModal}) => {
-                return (
-                    <HighlightsModal
-                        article={getItem()}
-                        closeModal={closeModal}
-                    />
-                );
-            })
-        ),
+        onTrigger: () => {
+            sdApi.highlights.fetchHighlights().then((res) => {
+                if ((res._items.length ?? 0) === 0) {
```

Comment:
are items `undefined` if there are no highlihgts? I think it should still be an empty array, thus applying a default isn't needed